Core/include/Acts/Propagator/EigenStepper.ipp

```diff
@@ -286,6 +291,18 @@ Acts::Result<double> Acts::EigenStepper<E>::step(
     }
   }
 
+  state.stepping.pathAccumulated += h;
+  ++state.stepping.nSteps;
+  state.stepping.nStepTrials += nStepTrials;
+
+  ++state.stepping.statistics.nSuccessfulSteps;
+  if (state.options.direction !=
+      Direction::fromScalarZeroAsPositive(initialH)) {
```

Comment:
This is checking if we're going opposite of the original stepping direction? Is this the most robust way to determine this? Maybe it doesn't matter for the statistic even if it isn't.

Reply:
I believe this is the only robust way to do this. Do you have an alternative way in mind?

`state.options.direction` represents the direction the propagator was configured to use. The sign of `initialH` will represent the actual direction we will go.

Separation wise this is not great. If we want to remove the propagator state from the stepper we need another mechanism but I would push this to the actual refactor.